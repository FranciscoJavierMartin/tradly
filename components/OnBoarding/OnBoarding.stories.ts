import OnBoarding from './OnBoarding.vue';
import slidesMock from './mock';

export default {
  title: 'OnBoarding',
  component: OnBoarding,
  args: {
    slides: slidesMock,
  },
  parameters: {
    layout: 'fullscreen',
  },
};

const Template = (args) => ({
  components: { OnBoarding },
  setup() {
    return { args };
  },
  template: '<OnBoarding v-bind="args"/>',
});

export const Basic = Template.bind({});
Basic.args = {};
