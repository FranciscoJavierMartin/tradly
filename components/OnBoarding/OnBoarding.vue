<template>
  <div class="c-on-boarding">
    <OnBoardingSlide
      :current-index="currentIndex"
      :amount-of-slides="slides.length"
      :slide="currentSlide"
      @next-slide="nextSlide"
    />
  </div>
</template>

<script setup lang="ts">
import { ref, computed, PropType } from 'vue';
import { Slide } from '@/components/OnBoardingSlide/types';
import OnBoardingSlide from '../OnBoardingSlide/OnBoardingSlide.vue';

const props = defineProps({
  slides: {
    type: Array as PropType<Slide[]>,
    default: () => [],
  },
});

const currentIndex = ref<number>(1);

const currentSlide = computed<Slide>(
  () => props.slides[currentIndex.value - 1]
);

function nextSlide(): void {
  currentIndex.value = (currentIndex.value % 3) + 1;
}
</script>

<style lang="scss">
@import '../../assets/styles/main.scss';

.c-on-boarding {
  background: linear-gradient($dark-green 50%, $white 50%);
  display: flex;
  flex-direction: column;
  height: 100vh;
  justify-content: flex-end;
  width: 100vw;
}
</style>
