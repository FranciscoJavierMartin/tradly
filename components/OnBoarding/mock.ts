export default [
  {
    img: '../../assets/imgs/on-boarding-step-1.svg',
    text: 'Empowering Artisans, Farmers & Micro Business',
  },
  {
    img: '../../assets/imgs/on-boarding-step-2.svg',
    text: 'Connecting NGOs, Social Enterprises with Communities',
  },
  {
    img: '../../assets/imgs/on-boarding-step-3.svg',
    text: 'Donate, Invest & Support infrastructure projects',
  },
];
