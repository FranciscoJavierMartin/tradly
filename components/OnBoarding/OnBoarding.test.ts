import '@testing-library/jest-dom';
import { describe, expect, test } from 'vitest';
import { render, screen } from '@testing-library/vue';
import OnBoarding from './OnBoarding.vue';

describe('OnBoarding.vue', () => {
  test('should render', () => {
    const { container } = render(OnBoarding);

    expect(
      screen.getByRole('heading', {
        name: /Empowering Artisans, Farmers & Micro Business/i,
      })
    ).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
