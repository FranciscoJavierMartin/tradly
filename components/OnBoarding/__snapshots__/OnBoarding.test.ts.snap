// Vitest Snapshot v1

exports[`OnBoarding.vue > should render 1`] = `
<div
  class="c-on-boarding"
>
  <img
    src="/assets/imgs/on-boarding-step-1.svg"
  />
  <h3>
    Empowering Artisans, Farmers & Micro Business
  </h3>
  <div
    class="steps"
  >
    <div
      class="active"
    />
    <div />
    <div />
  </div>
  <button>
    Next
  </button>
</div>
`;
